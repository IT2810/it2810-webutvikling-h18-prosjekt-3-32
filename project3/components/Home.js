import React from 'react';
import { StyleSheet, Text, View, List } from 'react-native';
import styles from "../stylesheets/Home.style.js";
import HomeListItem from "./HomeListItem.js";

export default class Home extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        eventsList: [
          {
            text: 'HMS-kurs',
            date: '2018-10-02'
          },
          {
            text: 'Bedpress med gutta as. Det blir fette najs. Kjørrr',
            date: '2018-10-04'
          }
        ],
        todoList: ['Bake pepperkaker', 'Fikse prosjekt3', 'Flæææ'],
        dailySteps: 100,
      };
    }

    render() {
        return (
          <View style={styles.homeContainer}>
            <View style={styles.calendarTodoContainer}>
                <View style={styles.nextEventsInfoBox}>
                  <Text style={styles.nextEventsInfoText}>{"Upcoming events"}</Text>
                </View>
                {
                  this.state.eventsList.map((item) => (
                    <HomeListItem text={item.text} date={item.date} key={item.text}/>
                  ))
                }
                <View style={styles.nextEventsInfoBox}>
                  <Text style={styles.nextEventsInfoText}>{"To do"}</Text>
                </View>
                {
                  this.state.todoList.map((item) => (
                    <HomeListItem text={item} key={item}/>
                  ))
                }
            </View>

            <View style={styles.showSteps}>
              <Text>{"Number of steps today: " + this.state.dailySteps}</Text>
            </View>
          </View>

        );
    }
};
