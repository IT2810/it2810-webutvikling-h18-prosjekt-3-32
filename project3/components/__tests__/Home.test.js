// Test that components mounts as expected

import React from 'react';
import Home from '../Home';
import renderer from 'react-test-renderer';
import MockAsyncStorage from "mock-async-storage";
import { AsyncStorage as storage } from "react-native";



test('Home.js renders correctly?', () => {
    const stepTree = renderer.create(<Home/>);
    expect(stepTree).toMatchSnapshot();
});

//Test to check if the function sortByDate works
test("check if the function sortByDate works", () => {
    //Creates instance of Home
    const testTodoListComponent = <Home/>;
    const homeComponent = renderer.create(testTodoListComponent).getInstance();

    //Creates an unsorted list
    const unsortedListByDate = [
        {key : 1, todoNr : "todo1", todoText: "test1", todoDate: "9999-12-24"},
        {key : 2, todoNr : "todo2", todoText: "test2", todoDate: "2000-12-24"}
    ];

    //Runs sortByDate which also triggers sortByDate
    const sortedList = homeComponent.sortByDate(unsortedListByDate);


    //Creates a sorted list
    const sortedListByDate = [
        {key : 2, todoNr : "todo2", todoText: "test2", todoDate: "2000-12-24"},
        {key : 1, todoNr : "todo1", todoText: "test1", todoDate: "9999-12-24"}
    ];

    //Checks if the state is the sorted list
    expect(sortedList).toEqual(sortedListByDate);
});

test("Functions using AsyncStorage", async () => {
    //Setting up mock Async
    const asyncMock = () => {
        const mockImpl = new MockAsyncStorage();
        jest.mock("AsyncStorage", () => mockImpl);
    };

    //Creates instance of Home
    const testTodoListComponent = <Home/>;
    const homeComponent = renderer.create(testTodoListComponent).getInstance();

    //Creates a testList
    const testTodoList = ["1", "test", "", false];
    const testAsyncList = [testTodoList[1], testTodoList[2], testTodoList[3]];

    //Initialize the mock for AsyncStorage
    asyncMock();

    await storage.setItem("todo"+testTodoList[0].toString(), JSON.stringify(testAsyncList));
    const value = await storage.getItem("todo1");
    const parsedValue = JSON.parse(value);

    const outList = [{
        key: "todo1",
        todoNr: "todo1",
        todoText: parsedValue[0],
        todoDate: parsedValue[1],
        done: parsedValue[2],
    }];

    homeComponent.refresh();

    expect(homeComponent.state.todoList).toEqual(outList);
});
