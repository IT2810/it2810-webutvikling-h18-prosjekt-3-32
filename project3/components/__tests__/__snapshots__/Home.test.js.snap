// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.js renders correctly? 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "height": "100%",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "height": "10%",
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "lineHeight": 50,
            "textAlign": "center",
          }
        }
      >
        Upcoming events
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "borderBottomWidth": 0.2,
          "display": "flex",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": "1.5%",
            "maxWidth": "70%",
          }
        }
      >
        HMS-kurs
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 13,
            "marginRight": "1.5%",
          }
        }
      >
        2018-10-02
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "borderBottomWidth": 0.2,
          "display": "flex",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": "1.5%",
            "maxWidth": "70%",
          }
        }
      >
        Bedpress med gutta as. Det blir fette najs. Kjørrr
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 13,
            "marginRight": "1.5%",
          }
        }
      >
        2018-10-04
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "height": "10%",
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "lineHeight": 50,
            "textAlign": "center",
          }
        }
      >
        To do
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "borderBottomWidth": 0.2,
          "display": "flex",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": "1.5%",
            "maxWidth": "70%",
          }
        }
      >
        Bake pepperkaker
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 13,
            "marginRight": "1.5%",
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "borderBottomWidth": 0.2,
          "display": "flex",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": "1.5%",
            "maxWidth": "70%",
          }
        }
      >
        Fikse prosjekt3
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 13,
            "marginRight": "1.5%",
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "borderBottomWidth": 0.2,
          "display": "flex",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": "1.5%",
            "maxWidth": "70%",
          }
        }
      >
        Flæææ
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 13,
            "marginRight": "1.5%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "height": "5%",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Number of steps last 24 hours: 100
    </Text>
  </View>
</View>
`;
